optimism/inverse/MechanicsInverse.py

```diff
@@ -0,0 +1,103 @@
+from collections import namedtuple
+
+from optimism.JaxConfig import *
```

Comment:
You shouldn't need this import. It's a legacy thing I'm tryin to remove. If it doesn't work without it, show me the error and I'll fix it.

Reply:
I happen to like import * from JaxConfig.  For me, its intent was to avoid having to constantly duplicate lots of lines like: from jax import grad, import blah, ... which I found to be annoying in every file.  It's an example where the DRY (dont repeat yourself) principal is in conflict with other principals.  For me, I like DRY here, but I know there is a trade off.  I understand that it pull in extra things, but that is essentially the point!  Without it, the tops of files are more verbose and it takes longer to get to the actual work of the file without such things.  If JaxConfig pull in things that may be duplicated or we don't want, we should delete them from JaxConfig.

Reply:
I understand what you mean, but there are better ways of doing that. The wildcard import is discouraged pretty much universally because it makes it so hard to understand where things are coming from, and names can collide and unintentionally overwrite the one you want. It's essentially like a GOTO - you'd have to follow all the `import *` instances to see which `grad` you're ultimately getting.  Even the Python project itself [doesn't allow it in their style](https://peps.python.org/pep-0008/#imports) any more. If you want to use the concretization/substitution idea, there are ways to do it by importing things in the `__init__.py` file for the project, which is similar, but forces you to be explicit (like your  MatrixClass = Armadillo::Matrix example).

Reply:
OK, can we basically put things like grad = jax.grad, etc. in __init__ then?  Basically things which are essentially always used and in the way we generally want to use them, so we can establish a standard way of doing it.  That was the intent of the .Config file.  In C++, the equivalent is basically #include "types.hpp", where types are put into the same namespace as the project on purpose.  That is essentially the same thing I want to do in python.  The issues is that putting it in __init__ puts in in EVERY file, where as "from optimism.types import *" is more selective.  So I don't really appreciate what the issue with that is.  We are not doing "from jax import *", we are including a types header.

Reply:
I think we should discuss this offline. Ryan, you can leave things as they are until we come to an agreement.

Reply:
Ok, go ahead and leave the wildcard import. We'll make a sweeping change in the future to get rid of them. Try to leave out the JaxConfig import in the future.